### ISSUE TITLE
Unable to start dev server in boilerplate

### ISSUE THREAD

**Issue by Hrishikesh-K** [netlify/cli#2958]
**Describe the bug**

The dev server doesn't start at all, throws the error: `◈ Cannot read property 'map' of undefined`. The error is thrown by the CLI and not the build tool as there's no building taking place. It's almost a blank website.

The problem occurs after updating to 4.3.0 and exists in 4.4.2 too. The last version where this worked fine was 4.2.3.

**To Reproduce**

1. Clone the repo: https://github.com/Hrishikesh-K/cli-issue. No need to install the Node Modules as there's nothing to build.
2. Navigate to the cloned repository
3. Run `netlify dev` (v4.3.0+)
4. See that the CLI exits with an error

**Configuration**

(Not completely relevant as occurs on other systems too)
```
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz
    Memory: 2.42 GB / 7.84 GB
  Binaries:
    Node: 16.2.0 - C:\Program Files\nodejs\node.EXE      
    npm: 7.19.0 - C:\Program Files\nodejs\npm.CMD  
```

**Expected behavior**

The server should start.

**CLI Output**

```
netlify dev        
◈ Netlify Dev ◈
◈ Ignored general context env var: LANG (defined in process)
◈ Cannot read property 'map' of undefined
```

**Additional context**

Reported by customer here: https://answers.netlify.com/t/cannot-start-netlify-locally-netlify-dev/41079/11

**Comment by coelmay** [comment]
Just to add to this, if `"gulp": "^4.0.2"`is removed from `devDependencies` dev will start.

**Comment by erezrokah** [comment]
Hi 👋 Sorry for the trouble.
I explained the cause in https://github.com/netlify/cli/pull/2970 which fixes this issue.

Removing the `gulp` dependency serves as a workaround as the CLI won't detect and try to use it.

**Comment by erezrokah** [comment]
Hello again, a fix for this has been released in [v5.0.1](https://github.com/netlify/cli/releases/tag/v5.0.1).
`v5` is a breaking change only if you are using Edge Handlers. See [here](https://github.com/netlify/cli/releases/tag/v5.0.0).

You can update via `npm install -g netlify-cli`

**Comment by coelmay** [comment]
Excellent.
